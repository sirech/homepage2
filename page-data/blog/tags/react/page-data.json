{
    "componentChunkName": "component---src-templates-blog-index-js",
    "path": "/blog/tags/react",
    "result": {"pageContext":{"siteMetadata":{"title":"Mario Fernandez","description":"This is the blog from Mario Fernandez, a Software Developer working for ThoughtWorks","author":"Mario Fernandez","url":"https://hceris.com","twitter":"@sirech"},"group":[{"frontmatter":{"layout":"post","title":"Applications Instead of Libraries","path":"/applications-instead-of-libraries/","categories":["Architecture","React","Micro frontends","JavaScript"],"date":"2021/11/20","draft":false,"description":"Using module federation to implement micro frontends"}},{"frontmatter":{"layout":"post","title":"From Redux to the Context API: A Practical Migration Guide","path":"/redux-to-context-practical-migration-guide/","categories":["React","Redux","Context API","Architecture","State Management"],"date":"2021/06/14","draft":false,"description":"Let's migrate a React application built with Redux to the Context API and see how it looks like"}},{"frontmatter":{"layout":"post","title":"Starting you new React application with Redux? Consider a simpler approach using the Context API","path":"/starting-a-new-app-with-redux-consider-context-api-first/","categories":["React","Redux","Context API","Architecture"],"date":"2021/05/11","draft":false,"description":"Traditionally, starting a new React application means adding Redux for state management. Let's talk about the Context API as an alternative"}},{"frontmatter":{"layout":"post","title":"Authorization Code Flow with PKCE (OAuth) in a React application","path":"/oauth-authorization-code-flow-pkce-for-react/","categories":["Auth0","OAuth","React"],"date":"2021/03/29","draft":false,"description":"SPAs implementing OAuth should pick Authorization Code Flow with PKCE for maximum security. Let's implement it in React with help of Auth0"}},{"frontmatter":{"layout":"post","title":"A directory structure for React projects","path":"/a-directory-structure-for-react-projects/","categories":["React","TypeScript","Code Organization"],"date":"2020/07/19","draft":false,"description":"How do you structure your components in a React application? I like to group mine around features close to the domain enforced by convention"}},{"frontmatter":{"layout":"post","title":"Fail a test in Jest if an unexpected network request happens","path":"/jest-fail-test-if-unexpected-network-request-happens/","categories":["React","Jest","Testing","react-testing-library","create-react-app","axios"],"date":"2019/12/08","draft":false,"description":"Automatically failing unit tests with Jest in React when network requests are made will make your tests more reliable and easier to maintain"}},{"frontmatter":{"layout":"post","title":"Custom Components in Formik","path":"/custom-components-in-formik/","categories":["React","Formik","react-hooks"],"date":"2019/08/15","draft":false,"description":"Formik is the new cool kid on the block for forms in react. Here is how to build custom input elements on top"}},{"frontmatter":{"layout":"post","title":"Migrating to Gatsby 2","path":"/migrating-to-gatsby2/","categories":["React","Gatsby"],"date":"2018/10/03","draft":false,"description":"Gatsby 2 is out! How did the migration go?"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 3","path":"/angular-from-react-part3/","categories":["JavaScript","React","Angular","TypeScript","angular-cli","redux"],"date":"2018/07/07","draft":false,"description":"Third part of my series on Angular, coming from a React background. This article focus on state management"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 2","path":"/angular-from-react-part2/","categories":["JavaScript","React","Angular","TypeScript","angular-cli"],"date":"2018/04/29","draft":false,"description":"Second part of my series on Angular, coming from a React background. This article focus on testing"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 1","path":"/angular-from-react-part1/","categories":["JavaScript","React","Angular","TypeScript","angular-cli"],"date":"2018/03/25","draft":false,"description":"First part of my series on Angular, coming from a React background. This article is about bootstrapping and TypeScript"}},{"frontmatter":{"layout":"post","title":"Migrating to Gatsby","path":"/migrating-to-gatsby/","categories":["React","Gatsby"],"date":"2018/03/18","draft":false,"description":"It all starts here. After using Create React App for a while, I decided to switch to Gatsby for my personal website"}}],"pathPrefix":"blog/tags/react","first":true,"last":true,"index":1,"pageCount":1,"additionalContext":{"siteMetadata":{"title":"Mario Fernandez","description":"This is the blog from Mario Fernandez, a Software Developer working for ThoughtWorks","author":"Mario Fernandez","url":"https://hceris.com","twitter":"@sirech"}}}},
    "staticQueryHashes": []}