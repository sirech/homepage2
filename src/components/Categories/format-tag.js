const _ = require('lodash')

export default (tag) => {
  return _.kebabCase(tag.toLowerCase())
}
