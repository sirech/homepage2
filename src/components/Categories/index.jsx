import React from 'react'
import PropTypes from 'prop-types'
import { addIndex, map } from 'ramda'

const renderCategories = (data) => {
  return addIndex(map)((item, i) => (
    <span className="badge badge-primary text-white mr-1" key={i}>
      {item}
    </span>
  ))(data)
}

const Categories = ({ categories }) => <>{renderCategories(categories)}</>

Categories.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.string).isRequired,
}
export default Categories
