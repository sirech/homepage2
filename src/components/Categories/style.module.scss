@import '../../layouts/colors';

.tag {
  background-color: map-get($theme-colors, 'primary');
  color: $white;

  &:hover {
    background-color: map-get($theme-colors, 'info');
  }
}
