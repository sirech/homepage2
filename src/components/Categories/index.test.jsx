import React from 'react'
import { render } from '@testing-library/react'
import Categories from './index'

describe('components', () => {
  describe('Categories', () => {
    it('renders correctly', () => {
      const { asFragment } = render(
        <Categories categories={['React', 'SpringBoot', 'SSL Certificates']} />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
