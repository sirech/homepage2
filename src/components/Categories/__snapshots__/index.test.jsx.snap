// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Categories renders correctly 1`] = `
<DocumentFragment>
  <a
    href="/blog/tags/react"
  >
    <span
      class="tag badge mr-1"
    >
      React
    </span>
  </a>
  <a
    href="/blog/tags/springboot"
  >
    <span
      class="tag badge mr-1"
    >
      SpringBoot
    </span>
  </a>
  <a
    href="/blog/tags/ssl-certificates"
  >
    <span
      class="tag badge mr-1"
    >
      SSL Certificates
    </span>
  </a>
</DocumentFragment>
`;
